Document revert strings config (#330)
foundry-rs/book: src/reference/forge/forge-build.md

See the [compiler input description documentation](https://docs.soliditylang.org/en/latest/using-the-compiler.html#compiler-input-and-output-json-description)
for more information on available settings (specifically `settings.optimizer.details`).

#### Revert Strings

> The revert strings is set to default, which means it does not inject compiler-generated revert strings and keeps user-supplied ones.

Reviewer: ```suggestion
You can control how revert strings are generated by the compiler. By default, only user supplied revert strings are included in the bytecode, but there are other options:
```

> There are three other options: 
> - Strip: Removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
> - Debug:  Injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
> - verboseDebug: Appends further information to user-supplied revert strings (not yet implemented).

Reviewer: ```suggestion
- `strip`: Removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
- `debug`:  Injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
- `verboseDebug`: Appends further information to user-supplied revert strings (not yet implemented).
```

> - Strip: It removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
> - Debug:  It injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
> - verboseDebug: It even appends further information to user-supplied revert strings (not yet implemented).

Reviewer 2: I would suggest removing the "It" at the beginning of these sentences
```suggestion
- Strip: Removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.
- Debug:  Injects strings for compiler-generated internal reverts, implemented for ABI encoders V1 and V2 for now.
- verboseDebug: Appends further information to user-supplied revert strings (not yet implemented).
```
